src/webgl/ShaderGenerator.js

```diff
@@ -1638,3 +1638,480 @@ export default shadergenerator;
+ *       color[0] *= this.brightness; // Tint red channel
+ *       return color;
+ *     });
+ *   });
+ * }
+ * function draw() {
+ *   background(220);
+ *   shader(myShader);
+ *   noStroke();
+ *   fill('teal');
+ *   box(100);
+ * }
+ * </code>
+ * </div>
+ */
+
+/**
+ * @function getPixelInputs
+ * @description
+ * Registers a callback to modify the properties of each fragment (pixel) before the final color is calculated in the fragment shader. This hook can be used inside <a href="#/p5/baseMaterialShader">baseMaterialShader()</a>.modify() and similar shader modify calls to change opacity or other per-pixel properties. The callback receives an object with the following properties:
+ * - `opacity`: a number between 0 and 1 for the pixel's transparency
+ * - (Other properties may be available depending on the shader.)
+ *
+ * Return the modified object to update the fragment.
+ *
+ * This hook is available in:
+ * - <a href="#/p5/baseMaterialShader">baseMaterialShader()</a>
+ * - <a href="#/p5/baseStrokeShader">baseStrokeShader()</a>
+ *
+ * @param {function} callback
+ *        A callback function which receives the fragment inputs object and should return it after making any changes.
+ *
+ * @example
+ * <div modernizr='webgl'>
+ * <code>
+ * let myShader;
+ * function setup() {
+ *   createCanvas(200, 200, WEBGL);
+ *   myShader = baseMaterialShader().modify(() => {
+ *     getPixelInputs(inputs => {
+ *       // Animate opacity based on x position
+ *       inputs.opacity = 0.5 + 0.5 * sin(inputs.texCoord.x * 10.0 + millis() * 0.002);
+ *       return inputs;
+ *     });
+ *   });
+ * }
+ * function draw() {
+ *   background(240);
+ *   shader(myShader);
+ *   lights();
+ *   noStroke();
+ *   fill('purple');
+ *   sphere(50);
+ * }
+ * </code>
+ * </div>
+ */
+
+/**
+ * @function shouldDiscard
+ * @description
+ * Registers a callback to decide whether to discard (skip drawing) a fragment (pixel) in the fragment shader. This hook can be used inside <a href="#/p5/baseStrokeShader">baseStrokeShader()</a>.modify() and similar shader modify calls to create effects like round points or custom masking. The callback receives a boolean:
+ * - `willDiscard`: true if the fragment would be discarded by default
+ *
+ * Return true to discard the fragment, or false to keep it.
+ *
+ * This hook is available in:
+ * - <a href="#/p5/baseStrokeShader">baseStrokeShader()</a>
+ *
+ * @param {function} callback
+ *        A callback function which receives a boolean and should return a boolean.
+ *
+ * @example
+ * <div modernizr='webgl'>
+ * <code>
+ * let myShader;
+ * function setup() {
+ *   createCanvas(200, 200, WEBGL);
+ *   myShader = baseStrokeShader().modify(() => {
+ *     shouldDiscard(willDiscard => {
+ *       // Discard fragments outside a circular region
+ *       return willDiscard || (inputs.position.x * inputs.position.x + inputs.position.y * inputs.position.y > 2500.0);
```

Comment:
`inputs` isn't defined here. Since we'd have to pass values between `getPixelInputs` and this, maybe this method is also one that we should `@private` until we have support for that.